typing/printtyp.ml

```diff
@@ -1022,8 +1024,9 @@ let rec tree_of_typexp sch ty =
         tree_of_typobject sch fi !nm
     | Tnil | Tfield _ ->
         tree_of_typobject sch ty None
-    | Tsubst ty ->
-        tree_of_typexp sch ty
+    | Tsubst _ ->
+        Otyp_stuff "<Tsubst>"
+          (* This case should not happen *)
```

Comment:
I would prefer an `assert false`: the outcome tree is user facing, no users should need to known what a `Tsubst` is.

Reply:
Sometimes we use Printtyp functions in the middle of the type-checker to debug things, but I guess we use `raw_type` rather than the higher-level functions. At least for `raw_type` I think it is important to handle intermediate states correctly.

Reply:
They need not see it: it is not going to happen, as many functions will now reject it anyway.
However, as @gasche pointed, one might want to use `Printtyp.type_expr` in the debugger for readability, and it is better not to have the debugger crashing because of an assertion.

Reply:
This seems sensible. However, the comment could be updated to: `This case should not happen except when debugging the compiler`.
Also, would it be too much to be extra-explicit with `<Error: unexpected Tsubst>`?

Reply:
I changed the comment but not the output. When this happens, this should actually not be unexpected.